cockatrice/src/soundengine.cpp

```diff
@@ -4,48 +4,76 @@
 #include <QAudioFormat>
 #include <QFile>
 #include <QBuffer>
+#include <QStringList>
+#if QT_VERSION > 0x050000
+    #include <QMediaPlayer>
+    #include <QUrl>
+#endif
 
 SoundEngine::SoundEngine(QObject *parent)
-    : QObject(parent), audio(0)
+: QObject(parent), audio(0)
 {
+    enabled = false;
     inputBuffer = new QBuffer(this);
     
     connect(settingsCache, SIGNAL(soundPathChanged()), this, SLOT(cacheData()));
     connect(settingsCache, SIGNAL(soundEnabledChanged()), this, SLOT(soundEnabledChanged()));
     cacheData();
     soundEnabledChanged();
-
-    lastTapPlayed = QDateTime::currentDateTime();
-    lastEndStepPlayed = QDateTime::currentDateTime();
-    lastAttackStepPlayed = QDateTime::currentDateTime();
 }
 
 void SoundEngine::cacheData()
 {
-    static const QStringList fileNames = QStringList()
-        << "end_step" << "tap" << "player_joined" << "attack";
-    for (int i = 0; i < fileNames.size(); ++i) {
-        QFile file(settingsCache->getSoundPath() + "/" + fileNames[i] + ".raw");
-        if(!file.exists())
-            continue;
-        file.open(QIODevice::ReadOnly);
-        audioData.insert(fileNames[i], file.readAll());
-        file.close();
-    }
+    fileNames = QStringList()
+        /*Phases*/
+        << "untap_step" << "upkeep_step" << "draw_step" << "main_1"
+        << "start_combat" << "attack_step" << "block_step" << "damage_step" << "end_combat"
+        << "main_2" << "end_step"
+        /*Game Actions*/
+        << "draw_card" << "play_card" << "tap_card" << "untap_card"
+        << "shuffle" << "roll_dice" << "life_change"
+        /*Player*/
+        << "player_join" << "player_leave" << "player_disconnect" << "player_reconnect" << "player_concede"
+        /*Spectator*/
+        << "spectator_join" << "spectator_leave"
+        /*Chat & UI*/
+        << "chat_mention" << "all_mention" << "private_message";
+
+    QStringList ext = QStringList() << ".mp4" << ".mp3" << ".wav";
+    #if QT_VERSION < 0x050000 //QT4
+        for (int i = 0; i < fileNames.size(); ++i) {
+            QFile file(settingsCache->getSoundPath() + "/" + fileNames[i] + ".wav");
+            if(!file.exists())
+                continue;
+            file.open(QIODevice::ReadOnly);
+            file.seek(44);
+            audioCache.insert(fileNames[i], file.readAll());
+            file.close();
+        }
+    #else //QT5
+        for (int i = 0; i < fileNames.size(); ++i) {
+            for (int j = 0; j < ext.size(); ++j) {
```

Comment:
Instead of the qt4 loop not having `j` why not make `ext` conditional on qt version and for qt4 only contain `wav`